### PR TITLE
Prevent the Formulas plugin depednant cell validator logic from validating cells outside of the schema.

### PR DESCRIPTION
**Pull request by jansiegel**
### Context
This issue fixes a problem described in #8215. 

<sup>[skip changelog]</sup>

### How has this been tested?
Added a test case for #8215.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #8215 

### Affected project(s):
- [x] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

### CODE REVIEW

**Review by aninde — APPROVED** [review]
I like such instant fixes 👍 
Both cases work, tested with trimmed row and hidden columns.

File: src/plugins/formulas/formulas.js [hunk 1]
```diff
@@ -436,16 +436,22 @@ export class Formulas extends BasePlugin {
     const changedCellsSet = new Set(changedCells.map(change => stringifyAddress(change)));
 
     dependentCells.forEach((change) => {
+      const { row, col } = change.address;
```
Comment (wszymanski) [inline src/plugins/formulas/formulas.js:439]: It seems that [address may be `undefined`](https://github.com/handsontable/handsontable/pull/8227/files#diff-75a3f65e47cad775c4d9cef1c98b6ef3ad0ad696c9aaa6b62eaf9fe45a09833cR448-R449). Will it work properly in that case? Shouldn't there be any extra conditional?

**Review by wszymanski — COMMENTED** [review]
I've just one question.

File: src/plugins/formulas/__tests__/formulas.spec.js [hunk 2]
```diff
@@ -2356,4 +2356,33 @@ describe('Formulas general', () => {
       });
     });
   });
+
+  it('should not try to validate cells outside of the table boundaries', () => {
```
Comment (budnix) [inline src/plugins/formulas/__tests__/formulas.spec.js:2360]: Can I ask you to move the validation test type to the `/src/plugins/formulas/__tests__/validation.spec.js` spec file?

File: src/plugins/formulas/formulas.js (hunk 1 above)
Comment (budnix) [reply src/plugins/formulas/formulas.js:439]: ⬆️  I agree with @wszymanski. Previously the coords destructing was secured by the `sheetId !== void 0`. In the case of the named expression, there can be unexpected behavior.

Comment (jansiegel) [reply src/plugins/formulas/formulas.js]: Does this make sense? fb9730b

File: src/plugins/formulas/__tests__/formulas.spec.js (hunk 2 above)
Comment (jansiegel) [reply src/plugins/formulas/__tests__/formulas.spec.js]: Sure thing! fb9730b

**Review by wszymanski — APPROVED** [review]

**Review by aninde — APPROVED** [review]
👍

**Pull request merged**